src/PolyType/Abstractions/Delegates.cs

```diff
@@ -32,11 +32,33 @@ namespace PolyType.Abstractions;
 /// <summary>
 /// Delegate representing a constructor that accepts a span of values.
 /// </summary>
-/// <typeparam name="T">The element type of the span parameters.</typeparam>
+/// <typeparam name="TKey">The type to be compared within the <typeparamref name="TElement"/>.</typeparam>
+/// <typeparam name="TElement">The element type of the span parameters.</typeparam>
 /// <typeparam name="TDeclaringType">The type of the value produced by the constructor.</typeparam>
 /// <param name="values">The span of values used to create the instance.</param>
+/// <param name="options">An optional set of parameters used to create the collection.</param>
 /// <returns>A newly constructed instance using the specified values.</returns>
-public delegate TDeclaringType SpanConstructor<T, TDeclaringType>(ReadOnlySpan<T> values);
+public delegate TDeclaringType SpanCollectionConstructor<TKey, TElement, TDeclaringType>(ReadOnlySpan<TElement> values, in CollectionConstructionOptions<TKey>? options = null);
```

Comment:
Can we avoid using nullable here?
```suggestion
public delegate TDeclaringType SpanCollectionConstructor<TKey, TElement, TDeclaringType>(ReadOnlySpan<TElement> values, in CollectionConstructionOptions<TKey> options = default);
```